VMEncryption/main/handle.py

```diff
@@ -270,6 +271,89 @@ def get_protected_settings():
     else:
         return protected_settings_str
 
+def update_encryption_settings_luks2_header(extra_items_to_encrypt=[]):
+    '''This function is used for CMK passphrse wrapping with new KEK URL and update metadata in LUKS2 header.'''
+    hutil.do_parse_context('UpdateEncryptionSettingsLuks2Header')
+    logger.log('Updating encryption settings LUKS-2 header')
+    # ensure cryptsetup package is still available in case it was for some reason removed after enable
+    try:
+        DistroPatcher.install_cryptsetup()
+    except Exception as e:
+        hutil.save_seq()
+        message = "Failed to update encryption settings with error: {0}, stack trace: {1}".format(e, traceback.format_exc())
+        hutil.do_exit(exit_code=CommonVariables.missing_dependency,
+                      operation='UpdateEncryptionSettingsLuks2Header',
+                      status=CommonVariables.extension_error_status,
+                      code=str(CommonVariables.missing_dependency),
+                      message=message)
+    try:
+        public_setting = get_public_settings()
+        encryption_config = EncryptionConfig(encryption_environment, logger)
+        extension_parameter = ExtensionParameter(hutil, logger, DistroPatcher, encryption_environment, get_protected_settings(), public_setting)
+        disk_util = DiskUtil(hutil=hutil, patching=DistroPatcher, logger=logger, encryption_environment=encryption_environment)
+        bek_util = BekUtil(disk_util, logger,encryption_environment)
+        device_items = disk_util.get_device_items(None)
+        for device_item in device_items:
+            device_item_path = disk_util.get_device_path(device_item.name)
+            if not disk_util.is_luks_device(device_item_path,None):
+                logger.log("Not a LUKS device, device path: {0}".format(device_item_path))
+                continue
+            #restoring the token data to type Azure_Disk_Encryption
```

Comment:
Add a comment saying this is necessary if we're resuming from previous attempt. Otherwise, it is no-op

Reply:
done